cfme/tests/infrastructure/test_kubevirt.py

```diff
@@ -3,23 +3,75 @@
 import pytest
 
 from cfme.infrastructure.provider.kubevirt import KubeVirtProvider
+from cfme.provisioning import do_vm_provisioning
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.generators import random_vm_name
+from cfme.utils.log import logger
 from cfme.utils.update import update
 
+
 pytestmark = [
-    pytest.mark.provider([KubeVirtProvider], scope="module")
+    pytest.mark.provider([KubeVirtProvider]),
+    pytest.mark.usefixtures('setup_provider')
 ]
 
 
-def test_k6t_provider_crud(provider):
-    """Tests provisioning via PXE
+@pytest.fixture
+def temp_vm(appliance, provider, provisioning):
+    template_name = provisioning['template']
+    vm_name = random_vm_name('k6tvm')
+    prov_data = {'catalog': {'vm_name': vm_name}}
+    vm = appliance.collections.infra_vms.instantiate(name=vm_name,
+                                                     provider=provider,
+                                                     template_name=template_name)
+    note = ('template {} to vm {} on provider {}'.format(template_name, vm_name, provider.key))
+    prov_data.update({
+        'request': {
+            'email': 'template_provisioner@example.com',
+            'first_name': 'Template',
+            'last_name': 'Provisioner',
+            'notes': note}})
+    view = navigate_to(vm, 'Provision')
+    view.form.fill_with(prov_data, on_change=view.form.submit_button)
+    view.flash.assert_no_error()
+    vm.wait_to_appear()
+    yield vm
+    vm.retire()
 
-    Metadata:
-        test_flag: v2v
-    """
-    provider.create()
+
+def test_k6t_provider_crud(provider):
 
     with update(provider):
         provider.name = fauxfactory.gen_alphanumeric() + '_updated'
 
     provider.delete(cancel=False)
     provider.wait_for_delete()
+
+
+@pytest.mark.parametrize('custom_prov_data',
+                         [{}, {'hardware': {'cpu_cores': '8', 'memory': '8192'}}],
+                         ids=['via_lifecycle', 'override_template_values'])
+def test_k6t_vm_crud(request, appliance, provider, provisioning, custom_prov_data):
+    vm_name = random_vm_name('k6tvm')
+    prov_data = {'catalog': {'vm_name': vm_name}}
+    provider.refresh_provider_relationships()
+    prov_data.update(custom_prov_data)
+    template = provisioning['template']
+    do_vm_provisioning(appliance, template, provider, vm_name, prov_data, request,
+                       smtp_test=False, wait=False)
+    logger.info('Waiting for cfme provision request for vm %s', vm_name)
```

Comment:
This is repeating a bunch of verification that's already in do_vm_provisioning method, and providing no additional validation that the VM exists, or matches the given parameters.  Please check out what's in do_vm_provisioning, and remove the duplicated code.

Reply:
Yes these are repeating lines, because doing `wait=True` would hit https://github.com/ManageIQ/integration_tests/blob/master/cfme/provisioning.py#L38
And `mgmt` is not yet in place for `kubevirt`